### DISCUSSION TITLE
Need help to contribute

### DISCUSSION THREAD

**Discussion by marionLa** [keystonejs/keystone#9046] [Questions] [answered]
Hi, 
I apologize for what may seem like a naive question, but I'm struggling to comprehend the process of contributing. I'm interested in helping with some issues, and I was wondering if you could guide me. I've gone through the discussion [7794](https://github.com/keystonejs/keystone/discussions/7794),  is this the correct approach? I've forked the project, but I'm not sure about how testing my changes. Do I require a starter project in a separate folder and then create symlinks on node-modules?
Thank you for your assistance and responses!
marion.

**Comment by marekryb** [comment, marked as answer]
Hi,

This project now uses `pnpm` package manager (https://pnpm.io/installation) with workspaces.
The way that worked for me so far is:
1) `git clone <your forked repo address>`
2) `pnpm install`
3) Select one of the examples (from `/examples` dir) that uses the thing that you want to modify.
4) In it's `package.json` modify keystone package version that you are modifying to `workspace:^`. Now this example will use source code with your changes instead of released package. Ex:
```
"@keystone-6/core": "workspace:^",
```
5) For additional validation run unit tests (`pnpm jest`). Some of them will not work in your local environment because they require database setup etc, so you can run just a subset through `pnpm jest packages/<package>/<nested path>`

**Reply by dcousens** [reply]
@marekryb this is a pretty great description.  Maybe we can add this to [CONTRIBUTING.md](https://github.com/keystonejs/keystone/blob/main/CONTRIBUTING.md), what was there a few months ago was pretty out of date by comparison to what you have written ^.

**Reply by marekryb** [reply]
Okay. I will add that to my backlog ~

**Reply by marionLa** [reply]
Thanks it's very clear for me now 👌 Good idea to add it on contributing.md ! 
Have a nice day
